core/src/mindustry/world/blocks/ConstructBlock.java

```diff
@@ -361,7 +361,7 @@ public void deconstruct(Unit builder, @Nullable CoreBuild core, float amount){
                 //add any leftover items that weren't obtained due to rounding errors
                 if(core != null && !state.rules.infiniteResources){
                     for(int i = 0; i < itemsLeft.length; i++){
-                        int target = Mathf.round(requirements[i].amount * state.rules.buildCostMultiplier * state.rules.deconstructRefundMultiplier);
+                        int target = Mathf.round(requirements[i].amount * state.rules.buildCostMultiplier * progress * state.rules.deconstructRefundMultiplier);
```

Comment:
I know the user account has now been deleted, but I would like a second opinion: If this code only runs when `progress <= 0` (assuming deconstructThreshold is 0 for pretty much all blocks), this makes `target` 0, which makes `remaining <= 0`, which means **nothing is ever refunded** in this part of the code, breaking all the itemsLeft code?